src/core/server/saved_objects/mappings/types.ts

```diff
@@ -17,47 +17,141 @@
  * under the License.
  */
 
-// FieldMapping isn't 1:1 with the options available,
-// modify as needed.
-export interface CoreFieldMapping {
+/**
+ * Internal representation of a saved object type mapping definition.
+ *
+ * @internal
+ */
+export interface SavedObjectsTypeMapping {
+  pluginId: string;
   type: string;
-  fields?: {
```

Comment:
I'm using a string instead of the `OpaquePluginId` here because the only usage of this field is to add details about the plugin name when two plugins are registering the same type, and using the String(pluginOpaqueId) outputs `Symbol(myId)`...